Panic reported in operator logs and operator pod restarted

**Description of problem:**

Rook csi ceph setup = latest rook + csi setup (after PR -https://github.com/rook/rook/pull/3324 was merged) 

In a 3 master + 3 worker(ceph nodes) setup, deleted a machine. The corresponding MON and OSD pods went into Pending state. But machineset added a new node to the cluster automatically. 

Then, after some time, a new mon(mon-d) came up on the new node( to keep the MON count=3 intact). But, as seen in operator logs,  when operator tried to destroy the failed OSD, it resulted in a panic and the operator pod restarted. 

**Log snip**
----------------
```
2019-06-26 13:27:55.071027 I | op-k8sutil: removing deployment rook-ceph-osd-1 if it exists
2019-06-26 13:27:55.080683 I | op-k8sutil: Removed deployment rook-ceph-osd-1
2019-06-26 13:27:55.085049 I | op-k8sutil: rook-ceph-osd-1 still found. waiting...
2019-06-26 13:27:57.089046 I | op-k8sutil: confirmed rook-ceph-osd-1 does not exist
2019-06-26 13:32:57.648512 I | op-osd: osd.1 is marked 'DOWN'
2019-06-26 13:32:57.648667 I | exec: Running command: ceph osd safe-to-destroy 1 --connect-timeout=15 --cluster=openshift-storage --conf=/var/lib/rook/openshift-storage/openshift-storage.config --keyring=/var/lib/rook/openshift-storage/client.admin.keyring --format json --out-file /tmp/291551664
2019-06-26 13:32:58.145233 I | op-osd: osd.1 is 'safe-to-destroy'
panic: runtime error: index out of range

goroutine 1717 [running]:
github.com/rook/rook/pkg/operator/ceph/cluster/osd.(*Monitor).handleOSDMarkedOut(0xc000accb00, 0x1, 0x16, 0xc0010a9e38)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/cluster/osd/health.go:119 +0x362
github.com/rook/rook/pkg/operator/ceph/cluster/osd.(*Monitor).osdStatus(0xc000accb00, 0xc000a38d80, 0x1)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/cluster/osd/health.go:98 +0x3b4
github.com/rook/rook/pkg/operator/ceph/cluster/osd.(*Monitor).Start(0xc000accb00, 0xc000baa720)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/cluster/osd/health.go:55 +0x130
created by github.com/rook/rook/pkg/operator/ceph/cluster.(*ClusterController).onAdd
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/cluster/controller.go:366 +0x124c

```
**Observations:**


1. The OSD couldn't be destroyed even after the timeout specified by operator.
Time when node was deleted(OSD and MON went down)  = 2019-06-26 13:14:54.562085

2. There was a panic in operator logs and the operator pod restarted abruptly. This led to loss of the previous logs.

3. As seen in ceph status, we even see the MGR as restarted, though the pod in "oc get pods" doesnt show the restart 

  services:
    mon: 3 daemons, quorum b,c,d (age 29m)
    mgr: a(active, since 22m)


**How reproducible:**
Tested once.

**Steps to Reproduce:**


1. Create a ceph based rook-csi cluster using the latest rook repo files (https://github.com/rook/rook) - master branch

2. Remove one of the machine 
  $ date; time oc delete machine/nberry-jun26-1-mwzjg-worker-us-east-2a-4ppjb -n openshift-machine-api; date

3. After some time, confirm that a new OCP node is spun up and a new machine is added.

4. Wait for MON timeout of 600 sec and confirm that a new MON is spinned on the new node( as required mon count in cluster=3).

5. Keep checking operator logs for any restarts or panic as showed above.



**Actual results:**

Operator pod reported panic and restarted.

**Expected results:**

The osd should be destoyed since it been OUT of the cluster for more than 30 mins

 osd: 3 osds: 2 up (since 40m), 2 in (since 30m)


**The OSD  is finally  not seen as part of below command,** 
sh-4.2# ceph osd status
+----+--------------------------------------------+-------+-------+--------+---------+--------+---------+----------------+
| id |                    host                    |  used | avail | wr ops | wr data | rd ops | rd data |     state      |
+----+--------------------------------------------+-------+-------+--------+---------+--------+---------+----------------+
| 0  | ip-10-0-149-134.us-east-2.compute.internal | 1025M | 97.9G |    0   |     0   |    1   |    16   |   exists,up    |
| 1  |                                            |    0  |    0  |    0   |     0   |    0   |     0   | autoout,exists |
| 2  | ip-10-0-161-198.us-east-2.compute.internal | 1025M | 97.9G |    0   |     0   |    1   |    90   |   exists,up    |
+----+--------------------------------------------+-------+-------+--------+---------+--------+---------+----------------+

**Version-Release number of selected component (if applicable):**

```
Ceph and rook version
-----------------------

sh-4.2# ceph version
ceph version 14.2.1 (d555a9489eb35f84f2e1ef49b77e19da9d113972) nautilus (stable)

sh-4.2# rook version
 rook: v1.0.0-154.g004f795


CSI version
-------------

$oc logs csi-rbdplugin-provisioner-0 -n openshift-storage -c csi-rbdplugin

I0626 09:40:26.830842       1 cephcsi.go:108] Starting driver type: rbd with name: rbd.csi.ceph.com
I0626 09:40:26.830922       1 rbd.go:104] Driver: rbd.csi.ceph.com version: 1.0.0

$ oc describe pod csi-rbdplugin-provisioner-0 -n openshift-storage|grep -i image
    Image:         quay.io/k8scsi/csi-provisioner:v1.2.0
    Image ID:      quay.io/k8scsi/csi-provisioner@sha256:0dffe9a8d39c4fdd49c5dd98ca5611a3f9726c012b082946f630e36988ba9f37
    Image:         quay.io/k8scsi/csi-attacher:v1.1.1
    Image ID:      quay.io/k8scsi/csi-attacher@sha256:e4db94969e1d463807162a1115192ed70d632a61fbeb3bdc97b40fe9ce78c831
    Image:         quay.io/k8scsi/csi-snapshotter:v1.1.0
    Image ID:      quay.io/k8scsi/csi-snapshotter@sha256:a49e0da1af6f2bf717e41ba1eee8b5e6a1cbd66a709dd92cc43fe475fe2589eb
    Image:         quay.io/cephcsi/cephcsi:canary
    Image ID:      quay.io/cephcsi/cephcsi@sha256:e832be9790bf12ab74c87217cce1dbd0a2416500e1d9a39af53b3fece414feac

Rook image
----------------------

$ oc describe pod rook-ceph-operator-5c6fd4b7db-rldcf -n openshift-storage|grep -i image
    Image:         rook/ceph:master
    Image ID:      docker.io/rook/ceph@sha256:4d0057e90c28a7bd8d3c3e9b13df40d0df1847567ef50a2a1e41dcea7ddb1d18
      ROOK_CSI_CEPH_IMAGE:                quay.io/cephcsi/cephcsi:canary
      ROOK_CSI_REGISTRAR_IMAGE:           quay.io/k8scsi/csi-node-driver-registrar:v1.1.0
      ROOK_CSI_PROVISIONER_IMAGE:         quay.io/k8scsi/csi-provisioner:v1.2.0
      ROOK_CSI_SNAPSHOTTER_IMAGE:         quay.io/k8scsi/csi-snapshotter:v1.1.0
      ROOK_CSI_ATTACHER_IMAGE:            quay.io/k8scsi/csi-attacher:v1.1.1

```

--- a/pkg/operator/ceph/cluster/osd/health.go
+++ b/pkg/operator/ceph/cluster/osd/health.go
@@ -23,6 +23,7 @@ import (
 	"github.com/rook/rook/pkg/clusterd"
 	"github.com/rook/rook/pkg/daemon/ceph/client"
 	"github.com/rook/rook/pkg/operator/k8sutil"
+	"k8s.io/apimachinery/pkg/api/errors"
 )
 
 const (
@@ -113,9 +114,14 @@ func (m *Monitor) handleOSDMarkedOut(outOSDid int) error {
 	if safeToDestroyOSD {
 		logger.Infof("osd.%d is 'safe-to-destroy'", outOSDid)
 		label := fmt.Sprintf("ceph-osd-id=%d", outOSDid)
-		dp, _ := k8sutil.GetDeployments(m.context.Clientset, m.clusterName, label)
-
-		if dp.Items != nil {
+		dp, err := k8sutil.GetDeployments(m.context.Clientset, m.clusterName, label)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil
+			}
+			return fmt.Errorf("failed to get osd deployment of osd id %d: %+v", outOSDid, err)
+		}
+		if len(dp.Items) != 0 {
 			if err := k8sutil.DeleteDeployment(m.context.Clientset, dp.Items[0].Namespace, dp.Items[0].Name); err != nil {
 				return fmt.Errorf("failed to delete osd deployment %s: %+v", dp.Items[0].Name, err)
 			}

